interpreter/cling/lib/Interpreter/PerfJITEventListener.cpp

```diff
@@ -0,0 +1,123 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author:  Guilherme Amadio <amadio@cern.ch>
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+//
+// This file implements a JITEventListener object that tells perf about JITted
+// symbols using perf map files (/tmp/perf-%d.map, where %d = pid of process).
+//
+// Documentation for this perf jit interface is available at:
+// https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/tools/perf/Documentation/jit-interface.txt
+//
+//------------------------------------------------------------------------------
+
+#ifdef __linux__
```

Comment:
```suggestion
#ifdef LLVM_ON_UNIX
```
Do we really need this for the cpp file and what makes this linux-specific?

Reply:
perf is part of the Linux kernel, and this is to support perf, so it's specific to Linux.